strapi/strapi
# fix(Filter by text field with "&" symbol doesn't work #20271)

What does it do?
Fixes a bug where filtering by text fields containing the "&" symbol fails due to improper encoding and decoding of query parameters.

Why is it needed?
Without this fix, users cannot correctly filter records by text fields containing special characters such as "&", which significantly hampers the usability of the filtering functionality.

How to test it?
Create an entity in a collection type with a text field containing the "&" symbol.
Example: Create an article titled "Test article & issue reproduce".
Go to the list of the collection type.
Use the "Filters" button to select the text field, "is" operand, and text with the "&" symbol.
Click "Add filter".
Verify that the filter is applied correctly and the appropriate records are displayed.

Related issue(s)/PR(s)
#20271
